### PR TITLE
adding ping identity provider

### PR DESCRIPTION
**Pull request by Mohammad-Khalid**
Hey, I'm adding this package as I'm thinking there are other folks who might be looking for a support Ping Identity as a provider in next-auth, easy and simple addition. Hope it helps

### CODE REVIEW

File: packages/next-auth/src/providers/ping-id.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/** @type {import(".").OAuthProvider} */
```
Comment (balazsorban44) [inline packages/next-auth/src/providers/ping-id.js:1]: Could you revert this please? `next-auth` does not accept new providers at this point. :pray:

File: packages/core/src/providers/ping-id.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/** @type {import(".").OAuthProvider} */
```
Comment (balazsorban44) [inline packages/core/src/providers/ping-id.js:1]: Could we convert this to TypeScript?

Also, make sure to document the code via JSDoc comments, have a look at https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/github.ts for a good example.

File: docs/docs/reference/05-oauth-providers/ping-id.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (balazsorban44) [inline docs/docs/reference/05-oauth-providers/ping-id.md:1]: As per https://github.com/nextauthjs/next-auth/pull/6614/files#r1096673092, this file can be deleted, as we would generate the API reference from the provided JSDoc comments.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks, added some comments.

File: packages/core/src/providers/ping-id.js (hunk 2 above)
Comment (Mohammad-Khalid) [reply packages/core/src/providers/ping-id.js:1]: Sure, I'll convert it into typescript and also document the code via JSDoc

**Comment by Mohammad-Khalid** [comment]
Hey @balazsorban44 the requested changes have been done, please review the changes and let me know if anything

File: packages/next-auth/src/providers/ping-id.js (hunk 1 above)
Comment (Mohammad-Khalid) [reply packages/next-auth/src/providers/ping-id.js]: I reverted it

File: packages/core/src/providers/ping-id.js (hunk 2 above)
Comment (Mohammad-Khalid) [reply packages/core/src/providers/ping-id.js]: Yes, it has been converted to TypeScript

File: docs/docs/reference/05-oauth-providers/ping-id.md (hunk 3 above)
Comment (Mohammad-Khalid) [reply docs/docs/reference/05-oauth-providers/ping-id.md]: The file has been deleted

**Comment by Mohammad-Khalid** [comment]
Hello @balazsorban44, I wanted to follow up on this and see if there have been any updates since I completed the changes you requested.

**Review by Gregor23021 — APPROVED** [review]

**Review by Gregor23021 — APPROVED** [review]

**Comment by onecrazygenius** [comment]
Any update on this?

File: packages/core/src/providers/ping-id.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import type { OAuthConfig, OAuthUserConfig } from "./index.js"
```
Comment (ThangHuuVu) [inline packages/core/src/providers/ping-id.ts:1]: Could you take the Auth0 provider as an example regarding JSDoc format and add doc for this one 🙏 https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/auth0.ts

File: packages/core/src/providers/ping-id.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  options: OAuthUserConfig<P>
+): OAuthConfig<P> {
+  return {
+    id: "ping-id",
+    name: "Ping Identity",
+    type: "oidc",
+    profile(profile) {
+      return {
+        id: profile.sub,
+        name: profile.name ?? profile.given_name + " " + profile.family_name,
```
Comment (ThangHuuVu) [inline packages/core/src/providers/ping-id.ts:62]: small nit:
```suggestion
        name: profile.name ?? `${ profile.given_name} ${profile.family_name}`,
```

**Review by ThangHuuVu — CHANGES_REQUESTED** [review]

**Comment by Flo-Bruno** [comment]
Is there any activity regarding this PR? @balazsorban44 @ThangHuuVu  @Mohammad-Khalid 

If not, I could re-open another PR and get this going if this is not moving.

**Review by Mohammad-Khalid — COMMENTED** [review]
LGTM

File: packages/core/src/providers/ping-id.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+ * If you think you found a bug in the default configuration, you can [open an issue](https://authjs.dev/new/provider-issue).
+ *
+ * Auth.js strictly adheres to the specification and it cannot take responsibility for any deviation from
+ * the spec by the provider. You can open an issue, but if the problem is non-compliance with the spec,
+ * we might not pursue a resolution. You can ask for more help in [Discussions](https://authjs.dev/new/github-discussions).
+ */
+
+export default function PingId<PingProfile>(
+  options: OIDCUserConfig<PingProfile>
+): OIDCConfig<PingProfile> {
```
Comment (balazsorban44) [inline packages/core/src/providers/ping-id.ts:61]:
```suggestion
export default function PingId(
  options: OIDCUserConfig<PingProfile>
): OIDCConfig<PingProfile> {
```

File: packages/core/src/providers/ping-id.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+ * ## Example
+ *
+ * ```ts
+ * import PingId from "@auth/core/providers/ping-id"
+ *
+ * ...
+ * providers: [
+ *  PingId({
+ *    clientId: process.env.AUTH_PING_ID_ID,
+ *    clientSecret: process.env.AUTH_PING_ID_SECRET,
```
Comment (balazsorban44) [inline packages/core/src/providers/ping-id.ts:43]:
```suggestion
 *    clientId: AUTH_PING_ID_ID,
 *    clientSecret: AUTH_PING_ID_SECRET,
```

**Review by falcowinkler — CHANGES_REQUESTED** [review]
@Mohammad-Khalid 
Can you please add a square svg image under `docs/public/img/providers/ping-id.svg`

Like this one should do just fine: 
ping-id

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**